### PR TITLE
Feature/get folder collaborations

### PR DESCRIPTION
**Pull request by valer-kovac**
GetCollaborationsAsync updated and GetFolderCollaborators_ValidResponse_ValidCollaborators, GetFolderCollaborators_ValidResponseWithGroups_ValidCollaborators UT

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @valer-kovac, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

**Comment by valer-kovac** [comment]
CLA signed

**Comment by boxcla** [comment]
Verified that @valer-kovac has just signed the CLA. Thanks, and we look forward to your contribution.

File: Box.V2.Test/BoxFilesManagerTest.cs [hunk 1]
```diff
@@ -189,6 +189,7 @@ public BoxFilesManagerTest()
 
             BoxFileRequest request = new BoxFileRequest()
             {
+                Id="5000948880",
```
Comment (coolboy) [inline Box.V2.Test/BoxFilesManagerTest.cs]: space between =

File: Box.V2/Managers/BoxFoldersManager.cs [hunk 2]
```diff
@@ -202,15 +202,22 @@ public BoxFoldersManager(IBoxConfig config, IBoxService service, IBoxConverter c
...
-        public async Task<BoxCollection<BoxCollaboration>> GetCollaborationsAsync(string id, List<string> fields = null)
+        /// <param name="id">Folder id.</param>
+        /// <param name="fields">Attribute(s) to include in the response.</param>
+        /// <param name="limit">The maximum number of items to return in a page. The default is 100 and the max is 1000.</param>
+        /// <param name="offset">The item at which to begin the response.</param>
+        /// <returns>A collection of collaboration objects are returned. If there are no collaborations on this folder, an empty collection will be returned.</returns>
+        public async Task<BoxCollection<BoxCollaboration>> GetCollaborationsAsync(string id, List<string> fields = null, int limit = 100, int offset = 0)
         {
             id.ThrowIfNullOrWhiteSpace("id");
+            limit.ThrowIfHigherThan(1000, "limit");
```
Comment (coolboy) [inline Box.V2/Managers/BoxFoldersManager.cs]: make this 1000 into a private or protected constant

**Comment by coolboy** [comment]
@valer-kovac conflict

**Comment by marek-vysoky** [comment]
@coolboy  Conflict resolved

**Comment by cburnette** [comment]
@coolboy Need to review this one.

File: Box.V2.Test/BoxResourceManagerTest.cs [hunk 3]
```diff
@@ -42,7 +44,8 @@ public BoxResourceManagerTest()
 
             _authRepository = new AuthRepository(_config.Object, _service, _converter, new OAuthSession("fakeAccessToken", "fakeRefreshToken", 3600, "bearer"));
         }
-        public static T CreateInstanceNonPublicConstructor<T>()
+
+		public static T CreateInstanceNonPublicConstructor<T>()
```
Comment (coolboy) [inline Box.V2.Test/BoxResourceManagerTest.cs]: spacing

File: Box.V2/Extensions/BoxExtensions.cs [hunk 4]
```diff
@@ -11,28 +11,66 @@ internal static class BoxExtensions
         /// Checks if the object is null 
         /// </summary>
         /// <typeparam name="T">Type of the object being checked</typeparam>
-        /// <param name="param"></param>
-        /// <param name="name"></param>
-        internal static T ThrowIfNull<T>(this T param, string name) 
+        /// <param name="param">Object being checked</param>
+        /// <param name="name">Object name</param>
+        internal static T ThrowIfNull<T>(this T param, string name) where T : class
```
Comment (coolboy) [inline Box.V2/Extensions/BoxExtensions.cs]: why add where T : class?

File: Box.V2/Extensions/BoxExtensions.cs [hunk 5]
```diff
@@ -11,28 +11,66 @@ internal static class BoxExtensions
...
+        }
+
+        /// <summary>
+        /// Checks if nulable enum  has value
+        /// </summary>
+        /// <typeparam name="T">Enum type</typeparam>
+        /// <param name="param">checked value</param>
+        /// <param name="name">Value name</param>
+        /// <returns>Input value if valid</returns>
+        internal static T? ThrowIfNullEnum<T>(this T? param, string name) where T : struct
```
Comment (coolboy) [inline Box.V2/Extensions/BoxExtensions.cs]: why not use ThrowIfNull above? assuming the where T : class restriction removed

File: Box.V2/Managers/BoxMetadataManager.cs [hunk 6]
```diff
@@ -173,7 +173,7 @@ public BoxMetadataManager(IBoxConfig config, IBoxService service, IBoxConverter
             foreach (BoxMetadataUpdate update in updates)
             {
                 update.Path.ThrowIfNullOrWhiteSpace("Path");
-                update.Op.ThrowIfNull("Op");
+                update.Op.ThrowIfNullEnum("Op");
```
Comment (coolboy) [inline Box.V2/Managers/BoxMetadataManager.cs]: ditto

**Comment by coolboy** [comment]
@cburnette reviewed, @marek-vysoky please address my comments

**Comment by marek-vysoky** [comment]
@coolboy I've removed ThrowIfNullEnum, please check again.

File: Box.V2.Test/BoxFilesManagerTest.cs [hunk 7]
```diff
@@ -59,27 +59,45 @@ public BoxFilesManagerTest()
...
-                }));
+                }))
+                .Callback<IBoxRequest>(r => boxRequest = r as BoxMultiPartRequest);
 
             var fakeFileRequest = new BoxFileRequest()
             {
                 Name = "test.txt",
-                ContentCreatedAt = DateTime.Now,
-                ContentModifiedAt = DateTime.Now,
+                ContentCreatedAt = new DateTime(2016, 8, 27),
```
Comment (coolboy) [inline Box.V2.Test/BoxFilesManagerTest.cs]: this test has timezone specific code and will fail on those machine that not on +02:00 timezone

File: Box.V2.Test/BoxResourceManagerTest.cs [hunk 8]
```diff
@@ -57,5 +74,14 @@ public static T CreateInstanceNonPublicConstructor<T>()
                                System.Threading.Thread.CurrentThread.CurrentCulture);
             return inst;
         }
+        public static bool AreJsonStringsEqual(string sourceJsonString, string targetJsonString)
```
Comment (coolboy) [inline Box.V2.Test/BoxResourceManagerTest.cs]: I think this will cause a duplicate function during merge?

**Comment by marek-vysoky** [comment]
@coolboy please, check this PR again?

**Comment by cburnette** [comment]
functionality exists in master

**Pull request closed without merging**